mantidproject/mantid
# TimeROI object in python

This exposes the new `mantid.kernel.TimeROI` object to python which will be used for implementing the new design for filtering events and logs #34794. 

**NOTE** do not merge before #34894 because it contains all of those changes as well.



**To test:**

```sh
ninja all
python -m pydoc mantid.kernel.TimeROI
```
see that there are docs for the exposed methods.

*There is no associated issue.*

*This does not require release notes* because it is not connected to anything user facing yet. Future PRs will add release notes.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
